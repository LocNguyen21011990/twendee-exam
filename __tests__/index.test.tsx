import { act, render, screen, cleanup } from '@testing-library/react';

import Home from '../src/pages/index';

afterEach(cleanup);

describe('Should render the app without crashing', () => {
  it('Renders the home page', async () => {
    await act(() => {
      render(<Home />)
    });
    expect(screen.getByText('User Table')).toBeInTheDocument()
  })
});

describe('Table matches snapshot', () => {
  it('Renders the home page', async () => {
    const mainHome = render(<Home />)
    const table = mainHome.getByTestId('mytable')
    expect(table).toMatchSnapshot()
  });
});