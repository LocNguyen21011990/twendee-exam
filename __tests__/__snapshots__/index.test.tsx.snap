// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the app without crashing Renders the home page 1`] = `
<table
  class="chakra-table css-0"
  data-testid="mytable"
  role="table"
>
  <thead
    class="css-0"
  >
    <tr
      class="css-0"
      role="row"
    >
      <th
        class="css-0"
        colspan="1"
        role="columnheader"
        style="cursor: pointer;"
        title="Toggle SortBy"
      >
        Full Name
        <span
          class="css-131wcsq"
        />
      </th>
      <th
        class="css-0"
        colspan="1"
        role="columnheader"
        style="cursor: pointer;"
        title="Toggle SortBy"
      >
        Username
        <span
          class="css-131wcsq"
        />
      </th>
      <th
        class="css-0"
        colspan="1"
        role="columnheader"
        style="cursor: pointer;"
        title="Toggle SortBy"
      >
        Thumbnail Icon
        <span
          class="css-131wcsq"
        />
      </th>
    </tr>
  </thead>
  <tbody
    class="css-0"
    role="rowgroup"
  />
</table>
`;

exports[`Table matches snapshot Renders the home page 1`] = `
<table
  class="chakra-table css-0"
  data-testid="mytable"
  role="table"
>
  <thead
    class="css-0"
  >
    <tr
      class="css-0"
      role="row"
    >
      <th
        class="css-0"
        colspan="1"
        role="columnheader"
        style="cursor: pointer;"
        title="Toggle SortBy"
      >
        Full Name
        <span
          class="css-131wcsq"
        />
      </th>
      <th
        class="css-0"
        colspan="1"
        role="columnheader"
        style="cursor: pointer;"
        title="Toggle SortBy"
      >
        Username
        <span
          class="css-131wcsq"
        />
      </th>
      <th
        class="css-0"
        colspan="1"
        role="columnheader"
        style="cursor: pointer;"
        title="Toggle SortBy"
      >
        Thumbnail Icon
        <span
          class="css-131wcsq"
        />
      </th>
    </tr>
  </thead>
  <tbody
    class="css-0"
    role="rowgroup"
  />
</table>
`;
